### PR TITLE
coder fixes

### PR DESCRIPTION
**Pull request by ryanbrinn**
islandora_solr

# What does this Pull Request do?

PHP Code Sniffer pass on islandora_solr

# How should this be tested?

Smoke testing Solr functionality in Drupal

### CODE REVIEW

File: src/Controller/DefaultController.php [hunk 1]
```diff
@@ -3,17 +3,44 @@
...
+class DefaultController extends ControllerBase implements ConfigFactoryInterface {
+
+  protected $moduleHandler;
+
+  protected $configFactory;
+
+  /**
+   * Constructor for dependency injection.
+   */
+  public function __construct(ModuleHandlerInterface $moduleHandler, ConfigFactoryInterface $configFactory) {
```
Comment (adam-vessey) [inline src/Controller/DefaultController.php]: This doesn't appear to be necessary? As in, access to both of these are already provided by [`ControllerBase`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Controller%21ControllerBase.php/class/ControllerBase/8.2.x):
* [`::moduleHandler()`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Controller%21ControllerBase.php/function/ControllerBase%3A%3AmoduleHandler/8.2.x) and
* [`::config()`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Controller%21ControllerBase.php/function/ControllerBase%3A%3Aconfig/8.2.x)

File: src/Controller/DefaultController.php [hunk 2]
```diff
@@ -3,17 +3,44 @@
...
 use Drupal\islandora_solr\IslandoraSolrResults;
 
+use Drupal\Component\Utility\Html;
+use Drupal\Component\Utility\Xss;
+
 /**
  * Default controller for the islandora_solr module.
  */
-class DefaultController extends ControllerBase {
+class DefaultController extends ControllerBase implements ConfigFactoryInterface {
```
Comment (adam-vessey) [inline src/Controller/DefaultController.php]: `ConfigFactoryInterface` doesn't quite appear appropriate? We don't actually implement the interface, do we? `ControllerBase` already implements `ContainerInjectionInterface`... isn't that all we use?...  Though as per below, we don't even need to do it on our own... what we grab is already available via methods added by various traits.

File: src/Form/IslandoraDateFilter.php [hunk 3]
```diff
@@ -71,7 +71,9 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
    */
   public function buildForm(array $form, FormStateInterface $form_state, array $elements = []) {
     global $_islandora_solr_queryclass;
-    extract($elements);
+    $facet_field = elements['facet_field'];
+    $form_key = elements['form_key'];
+    $datepicker_range = elements['datepicker_range'];
```
Comment (adam-vessey) [inline src/Form/IslandoraDateFilter.php]: Assuming these three should be referring to `$elements`?

File: src/Form/IslandoraDateFilter.php [hunk 4]
```diff
@@ -112,13 +114,17 @@ public function buildForm(array $form, FormStateInterface $form_state, array $el
...
+              $from_default = (strpos($filter_array[0], '*') !== FALSE) ? '*' : \Drupal::getContainer()->get('date.formatter')->format($from_unix, 'custom', $format, 'UTC');
+              // @codingStandardsIgnoreEnd
             }
             else {
               $from_default = NULL;
             }
             if ($to_unix !== FALSE) {
-              $to_default = (strpos($filter_array[1], '*') !== FALSE) ? '*' : format_date($to_unix, 'custom', $format, 'UTC');
+              // @codingStandardsIgnoreStart
+              $to_default = (strpos($filter_array[1], '*') !== FALSE) ? '*' : \Drupal::getContainer()->get('date.formatter')->format($to_unix, 'custom', $format, 'UTC');
```
Comment (adam-vessey) [inline src/Form/IslandoraDateFilter.php]: Should... we just do DI to get the `date.formatter` service?

File: islandora_solr_config/src/Form/IslandoraSolrConfigAdminRssSettings.php [hunk 5]
```diff
@@ -22,8 +22,8 @@ public function getFormId() {
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
     // Get variables.
-    $rss_item = \Drupal::config('islandora_solr_config.settings')->get('islandora_solr_config_rss_item');
-    $rss_channel = \Drupal::config('islandora_solr_config.settings')->get('islandora_solr_config_rss_channel');
+    $rss_item = self::config('islandora_solr_config.settings')->get('islandora_solr_config_rss_item');
+    $rss_channel = self::config('islandora_solr_config.settings')->get('islandora_solr_config_rss_channel');
```
Comment (adam-vessey) [inline islandora_solr_config/src/Form/IslandoraSolrConfigAdminRssSettings.php]: Static references to non-static... should probably be `$this->config()`, to call [the method from `FormBase`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Form%21FormBase.php/function/FormBase%3A%3Aconfig/8.2.x)? (same deal elsewhere, with `self::config()`)

File: src/Form/IslandoraRangeSlider.php [hunk 6]
```diff
@@ -29,8 +29,10 @@ public function getFormId() {
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state, array $elements = []) {
-    global $_islandora_solr_queryclass;
-    extract($elements);
+    $gap = elements['gap'];
+    $facet_field = elements['facet_field'];
+    $form_key = elements['form_key'];
+    $slider_color = elements['slider_color'];
```
Comment (adam-vessey) [inline src/Form/IslandoraRangeSlider.php]: Same deal, should probably be referring to `$elements`.

File: src/Form/IslandoraDateFilter.php (hunk 4 above)
Comment (ryanbrinn) [reply src/Form/IslandoraDateFilter.php]: I believe this would require more refactoring than is warranted here.

Comment (ryanbrinn) [reply src/Form/IslandoraDateFilter.php]: add an XXX

File: islandora_solr_config/src/Form/IslandoraSolrConfigAdminTableProfileSettings.php [hunk 7]
```diff
@@ -29,13 +29,13 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       'islandora_solr_table_profile_display_row_no' => [
         '#type' => 'checkbox',
         '#title' => $this->t('Display Row Numbers?'),
-        '#default_value' => \Drupal::config('islandora_solr_config.settings')->get('islandora_solr_table_profile_display_row_no'),
+        '#default_value' => self::config('islandora_solr_config.settings')->get('islandora_solr_table_profile_display_row_no'),
```
Comment (adam-vessey) [inline islandora_solr_config/src/Form/IslandoraSolrConfigAdminTableProfileSettings.php]: More static references to non-statics... should possibly do something of a grep for 'em?

**Review by adam-vessey — COMMENTED** [review]
Looks like I forgot to submit this last time...

**Pull request closed without merging**